elytron/WFLY-14984-encrypted-expression-for-client-config.adoc

```diff
@@ -0,0 +1,164 @@
+= Add support for encrypted expressions in the wildfly-config.xml
+:author:            Prarthona Paul
+:email:             prpaul@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+WildFly includes a common configuration framework that can be used to configure multiple clients, rather than relying on each client having their own configuration file. The configuration can be defined using a `wildfly-config.xml` file, where there are multiple configuration sections containing attributes whose values determine the configuration of the WildFly clients. 
+
+Attribute values for client configuration can include sensitive information. And while credential references can be used for some of them, others still need to be defined using plaintext. As a result, those attributes are still exposed and make the application vulnerable. Additionally, the password for the credential store needs to be defined as clear-text, which is not ideal. 
+
+This feature adds support for encrypted expressions for attributes in `wildfly-config.xml` file, which allows users to avoid specifying sensitive information using plaintext. Instead, encrypted expressions can be used to specify these information. A configuration section named `encrypted-expression` would be added, which would reference a `secret-key-credential-store` to extract encrypted expressions. It would also have a resolver, which would be backed by the secret-key-credential-store. The `secret-key-credential-store` can be created either using the `elytron-tool` or using the Elytron subsystem on the WildFly server.
```

Comment:
This is probably just a copy/paste thing but since wildfly-config.xml is something that's used on the client side, we shouldn't be mentioning the Elytron subsystem here.

Reply:
I was thinking in case the user is doing some server configuration as well like this blog does: https://wildfly-security.github.io/wildfly-elytron/blog/ejb-over-tls/
In these cases, we can also use the management cli to create the credential store as opposed to using the elytron-tool. 
WDYT? Could this be a possible use case?

Reply:
In the linked blog post, it looks like the credential store is created using elytron-tool.sh. I don't see subsystem commands being used to create it.

The management CLI is used for server config so I think mentioning it just makes things a bit confusing since this RFE is about the client side.